packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -810,7 +840,19 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
                 }
                 case 'Shift': {
                     if (!this.isRightClick) {
-                        this.setState({ cursor: 'crosshair', shiftKey: true });
+                        if (key.ctrlKey) {
+                            this.setState({ cursor: 'default' });
+                        } else {
+                            this.setState({ cursor: 'crosshair', shiftKey: true });
+                        }
+                    }
+                    break;
+                }
+                case 'Control': {
+                    if (key.shiftKey) {
+                        this.setState({ cursor: 'default' });
+                    } else if (!this.state.shiftKey) {
+                        this.setState({ cursor: 'grabbing', ctrlKey: true });
```

Comment:
These keys interfere with the cursor when we are already panning or selecting. They could be ignored when isPanning is true, but we have to find another condition for selection, unless we also add a isSelecting state?

Reply:
Maybe could use this.mouseIsDown...

Reply:
Yeah, I used the mouseIsDown check in `private onMouseMove()`, so here we can leave as it is.

Reply:
The cursor changes if you press and release the modifier keys during an operation (selecting or panning). I think the cursor should be fixed to the current operation in progress.

While selecting, holding the Ctrl key will temporarily stop the selection from extending to the mouse position.

Reply:
You are right. I used your suggestion of `isSelection`, as `mouseIsDown` was not enough to solve this.